comp/core/settings/settingsimpl/settingsimpl.go

```diff
@@ -0,0 +1,221 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+// Package settingsimpl implements the interface for the settings component
+package settingsimpl
+
+import (
+	"encoding/json"
+	"html"
+	"net/http"
+	"sync"
+
+	"go.uber.org/fx"
+	"gopkg.in/yaml.v2"
+
+	"github.com/DataDog/datadog-agent/comp/core/log"
+	"github.com/DataDog/datadog-agent/comp/core/settings"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/config/model"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"github.com/DataDog/datadog-agent/pkg/util/scrubber"
+)
+
+// Module defines the fx options for this component.
+func Module() fxutil.Module {
+	return fxutil.Component(
+		fx.Provide(newSettings),
+	)
+}
+
+type provides struct {
+	fx.Out
+
+	Comp settings.Component
+}
+
+type dependencies struct {
+	fx.In
+
+	Log      log.Component
+	Settings []settings.RuntimeSetting `group:"runtime_setting"`
+}
+
+type settingsRegistry struct {
+	rwMutex            sync.RWMutex
+	registeredSettings map[string]settings.RuntimeSetting
+	log                log.Component
+}
+
+// RuntimeSettings returns all runtime configurable settings
+func (s *settingsRegistry) RuntimeSettings() map[string]settings.RuntimeSetting {
+	return s.registeredSettings
```

Comment:
I think we should return a copy of the map here to avoid the caller from modifying the internal map.

Reply:
I changed the code to return a copy. The problem is that if the values are pointer the copy is useless